Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp

```diff
@@ -1311,8 +1312,13 @@ MediaPlayerEnums::SupportsType SourceBufferParserWebM::isContentTypeSupported(co
 #if ENABLE(VP9)
         if (codec.startsWith("vp09"_s) || codec.startsWith("vp08"_s) || equal(codec, "vp8"_s) || equal(codec, "vp9"_s)) {
 
-            if (!isVP9DecoderAvailable())
-                return MediaPlayerEnums::SupportsType::IsNotSupported;
+            if (codec.startsWith("vp09"_s) || equal(codec, "vp9"_s)) {
+                if (!isVP9DecoderAvailable())
+                    return MediaPlayerEnums::SupportsType::IsNotSupported;
+            } else {
```

Comment:
else if { ... }

Reply:
Are you sure? In this case I like this structure better than "else if". The two sides of the if/else are analogous so it’s nice that they are similarly formatted.

Reply:
> Are you sure? In this case I like this structure better than "else if". The two sides of the if/else are analogous so it’s nice that they are similarly formatted.

We are here because it's the vp8 codecs (as the test to enter this block was :
```
if (codec.startsWith("vp09"_s) || codec.startsWith("vp08"_s) || equal(codec, "vp8"_s) || equal(codec, "vp9"_s))
```

Reply:
> How about something like this?
> 
> ```c++
>         bool isVP9 = codec.startsWith("vp09"_s) || equal(codec, "vp9"_s);
>         bool isVP8 = codec.startsWith("vp08"_s) || equal(codec, "vp8"_s);
>         if (isVP9 || isVP8) {
> 
>             if (isVP9 && !isVP9DecoderAvailable())
>                 return MediaPlayerEnums::SupportsType::IsNotSupported;
>             else if (isVP8 && !isVP8DecoderAvailable())
>                 return MediaPlayerEnums::SupportsType::IsNotSupported;
> 
>             // ...
>         }
> ```

Yes, this is much nicer thank you.
clearer, and remove unnecessary string comparisons.